tensorflow_quantum/core/src/circuit_parser_qsim_test.cc

```diff
@@ -0,0 +1,480 @@
+  auto reference = qsim::Cirq::FSimGate<float>::Create(0, 0, 1, theta, phi);
+  Program program_proto;
+  Circuit* circuit_proto = program_proto.mutable_circuit();
+  circuit_proto->set_scheduling_strategy(circuit_proto->MOMENT_BY_MOMENT);
+  Moment* moments_proto = circuit_proto->add_moments();
+
+  // Add gate.
+  Operation* operations_proto = moments_proto->add_operations();
+  Gate* gate_proto = operations_proto->mutable_gate();
+  gate_proto->set_id("FSIM");
+
+  // Set the args.
+  google::protobuf::Map<std::string, Arg>* args_proto =
+      operations_proto->mutable_args();
+  (*args_proto)["theta"] = MakeArg("alpha");
+  (*args_proto)["theta_scalar"] = MakeArg(0.5);
+  (*args_proto)["phi"] = MakeArg("beta");
+  (*args_proto)["phi_scalar"] = MakeArg(0.2);
+
+  // Set the qubits.
+  Qubit* qubits_proto = operations_proto->add_qubits();
+  qubits_proto->set_id("0");
+  qubits_proto = operations_proto->add_qubits();
+  qubits_proto->set_id("1");
+
+  QsimCircuit test_circuit;
+  std::vector<qsim::GateFused<QsimGate>> fused_circuit;
+  SymbolMap symbol_map = {{"alpha", std::pair<int, float>(0, 2 * theta)},
+                          {"beta", std::pair<int, float>(1, 5 * phi)}};
+
+  // Test symbol resolution.
+  ASSERT_EQ(QsimCircuitFromProgram(program_proto, symbol_map, 2, &test_circuit,
+                                   &fused_circuit),
+            tensorflow::Status::OK());
+  AssertTwoQubitEqual(test_circuit.gates[0], reference);
+
+  symbol_map.clear();
+  test_circuit.gates.clear();
+  fused_circuit.clear();
+  (*args_proto)["theta"] = MakeArg(theta);
+  (*args_proto)["theta_scalar"] = MakeArg(1.0);
+  (*args_proto)["phi"] = MakeArg(phi);
+  (*args_proto)["phi_scalar"] = MakeArg(1.0);
+
+  // Test float values only.
+  ASSERT_EQ(QsimCircuitFromProgram(program_proto, symbol_map, 2, &test_circuit,
+                                   &fused_circuit),
+            tensorflow::Status::OK());
+  AssertTwoQubitEqual(test_circuit.gates[0], reference);
+
+  test_circuit.gates.clear();
+  fused_circuit.clear();
+  args_proto->erase(args_proto->find("theta"));
+
+  // Test case where proto arg missing.
+  ASSERT_EQ(QsimCircuitFromProgram(program_proto, symbol_map, 2, &test_circuit,
+                                   &fused_circuit),
+            tensorflow::Status(tensorflow::error::INVALID_ARGUMENT,
+                               "Could not find arg: theta in op."));
+
+  test_circuit.gates.clear();
+  fused_circuit.clear();
+  (*args_proto)["theta"] = MakeArg("alpha");
+  symbol_map.clear();
+
+  // Test case where symbol value not present in resolver.
+  ASSERT_EQ(
+      QsimCircuitFromProgram(program_proto, symbol_map, 2, &test_circuit,
+                             &fused_circuit),
+      tensorflow::Status(tensorflow::error::INVALID_ARGUMENT,
+                         "Could not find symbol in parameter map: alpha"));
+}
+
+TEST(QsimCircuitParserTest, PhasedISwapTest) {
+  float exponent = 0.1234;
+  float phase_exponent = 0.4567;
```

Comment:
PXP(PhasedXPowGate) is missed. I think the PXP(PhasedXPowGate) can be tested inside here with this PISP. Could you?

Reply:
Added.